init/index.js

```diff
@@ -7,9 +7,28 @@ function generatorPatternPack() {
   var generator;
 
   function constructor() {
-    generators.NamedBase.apply(this, arguments);
+    generators.Base.apply(this, arguments);
     generator = generator || this;
-    generator.options.name = generator.name;
+    generator.argument('name', { required: false });
+  }
+
+  function promptForName() {
```

Comment:
Added the promptForName so that if the user forgets to provide a name, they are prompted instead of an error occurring.